lib/errors.js

```diff
@@ -1,39 +1,73 @@
+'use strict';
+
+var util = require('util');
+
+/**
+ * The base class for all thinky errors
+ *
+ * @constructor
+ * @alias Error
+ */
+var ThinkyError = function(message) {
+  this.name = 'ThinkyError';
+  this.message = message || 'ThinkyError';
+  Error.captureStackTrace && Error.captureStackTrace(this, this.constructor);
+};
+util.inherits(ThinkyError, Error);
+
 /**
  * DocumentNotFound error which is returned when `get` returns `null`.
  * "Extends" Error
  */
-function DocumentNotFound(message) {
-  Error.captureStackTrace(this, DocumentNotFound);
+var DocumentNotFoundError = function(message) {
+  ThinkyError.call(this, message);
+  this.name = "DocumentNotFoundError";
```

Comment:
Why did you move it in the constructor? It's technically not muting, so there's no need to recreate the name every time.